src/main/java/com/google/graphgeckos/dashboard/storage/DataRepository.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+/**
+ * An interface for providing unified I/O to the storage component.
+ */
+public interface DataRepository {
+  /**
+   * Creates a new database entry about a te metadata of a commit. If there already
+   * is an entry with the same commit hash, ignores the request.
+   *
+   * @param entryData a ParsedGitData instance, must have a non-null "commitHash" field
+   * @throws IllegalArgumentException if entryData is null or doesn't have a
+   *      "commitHash" field
+   */
+  void createEntry(ParsedGitData entryData);
```

Comment:
Should the throws keyword also be added here in this line if it's mentioned in the comment, just like in line 65?

Reply:
It's upcoming in the next commit. I forgot to update a few details about what exceptions can occur inside each method, and I'm working on that right now.